src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -65,6 +65,22 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
                     {
                         ConfigureTempApiHashKey(builder, settings.Authentication);
                     }
+
+                    // User-specified configuration file path is considered highest precedence, but does NOT override other configuration sources
+                    string userFilePath = settings.UserProvidedConfigFilePath;
+
+                    if (File.Exists(userFilePath) && Path.GetExtension(userFilePath).ToLower().Equals(".json"))
+                    {
+                        try
+                        {
+                            File.OpenRead(userFilePath).Dispose(); // If this succeeds, we have read-permissions
+                            builder.AddJsonFile(userFilePath, optional: true, reloadOnChange: true);
+                        }
+                        catch (Exception)
+                        {
+                            // Don't currently log to tell the user that their configuration file was not loaded
+                        }
+                    }
```

Comment:
Do we want to have something that tells users here that their path either 1) didn't exist or 2) wasn't a json file?